integration-tests/testutils/report/parser.go

```diff
@@ -40,43 +42,52 @@ var (
 	skipRegexp     = regexp.MustCompile(fmt.Sprintf(commonPattern, "SKIP") + skipPatternSufix)
 )
 
-// ParserReporter is a type implementing io.Writer that
-// parses the input data and sends the results to the Next
-// reporter
+// Statuser reports the status of a test.
+type Statuser interface {
+	Status(subunit.Event) error
+}
+
+// SubunitV2ParserReporter is a type that parses the input data
+// and sends the results as a test status.
 //
 // The input data is expected to be of the form of the textual
 // output of gocheck with verbose mode enabled, and the output
-// will be of the form defined by the subunit format before
-// the binary encoding. There are constants reflecting the
-// expected patterns for this texts.
+// will be of the form defined by the subunit v2 format There
+// are constants reflecting the expected patterns for this texts.
 // Additionally, it doesn't take  into account the SKIPs done
 // from a SetUpTest method, due to the nature of the snappy test
 // suite we are using those for resuming execution after a reboot
 // and they shouldn't be reflected as skipped tests in the final
 // output. For the same reason we use a special marker for the
 // test's announce.
-type ParserReporter struct {
-	Next io.Writer
+type SubunitV2ParserReporter struct {
+	statuser Statuser
+}
+
+// NewSubunitV2ParserReporter returns a new ParserReporter that sends the report to the
+// writer argument.
+func NewSubunitV2ParserReporter(writer io.Writer) *SubunitV2ParserReporter {
+	return &SubunitV2ParserReporter{statuser: &subunit.StreamResultToBytes{Output: writer}}
 }
 
-func (fr *ParserReporter) Write(data []byte) (n int, err error) {
-	var outputStr string
+func (fr *SubunitV2ParserReporter) Write(data []byte) (int, error) {
+	var err error
 
 	if matches := announceRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
-		outputStr = fmt.Sprintf("test: %s\n", matches[1])
-
+		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "exists"})
 	} else if matches := successRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
-		outputStr = fmt.Sprintf("success: %s\n", matches[1])
-
+		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "success"})
 	} else if matches := failureRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
-		outputStr = fmt.Sprintf("failure: %s\n", matches[1])
-
+		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "fail"})
 	} else if matches := skipRegexp.FindStringSubmatch(string(data)); len(matches) == 3 {
-		outputStr = fmt.Sprintf("skip: %s [\n%s\n]\n", matches[1], matches[2])
+		err = fr.statuser.Status(subunit.Event{
+			TestID:    matches[1],
+			Status:    "skip",
+			FileName:  "reason",
+			FileBytes: []byte(matches[2]),
+			MIME:      "text/plain;charset=utf8",
+		})
 	}
 
-	outputByte := []byte(outputStr)
-	n = len(outputByte)
-	fr.Next.Write(outputByte)
-	return
+	return 0, err
```

Comment:
I didn't know what to do with the number of bytes written. As we are not using them anywhere, the smallest possible change was to just return 0.
Alternatives might be:

    define an io.Writer that also records the number of bytes written on each call to write, so we can a return them later. This seems weird.
    modify subunit.Status to return the total number of bytes written. I don't like this because it doesn't sound like the job of Status to return the number of bytes. Any suggestions?